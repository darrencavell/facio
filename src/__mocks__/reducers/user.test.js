import reducer , { initialState } from '../../reducers/user.js'

describe('reducer user redux', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })
  it('should handle LOGIN', () => {
    expect(
      reducer(initialState,
      {
        type: 'LOGIN',
        payload: {
          tokn: 'token'
        }
      })
    ).toMatchSnapshot()
  })
  it('should handle LOGIN_LOADING', () => {
    expect(
      reducer(initialState,
      {
        type: 'LOGIN_LOADING',
        payload: true
      })
    ).toMatchSnapshot()
  })

  it('should handle LOGIN_ERROR', () => {
    expect(
      reducer(initialState,
      {
        type: 'LOGIN_ERROR',
        payload: {
          error: 'Username not found'
        }
      })
    ).toMatchSnapshot()
  })
})