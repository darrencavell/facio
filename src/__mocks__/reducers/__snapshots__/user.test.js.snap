// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action redux should handle ADD_TRIP 1`] = `
Object {
  "add": Object {
    "response": Object {
      "data": Object {
        "created_at": "2020-08-18 17:12:00",
        "driver_name": "John Doe",
        "end_station": "Semarang",
        "eta": "2020-08-20 00:00:00",
        "start_station": "Surabaya",
        "status": "Transit",
      },
    },
  },
  "lists": Object {},
}
`;

exports[`action redux should handle ADD_TRIP_LOADING 1`] = `
Object {
  "add": Object {
    "loading": true,
  },
  "lists": Object {},
}
`;

exports[`action redux should handle LOGIN 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`action redux should handle LOGIN_ERROR 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`action redux should handle LOGIN_LOADING 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`action redux should handle TRIP_LIST 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": Object {
      "_id": "5f521f25d899cd03e8b4de79",
      "created_at": "2020-08-18 17:12:00",
      "driver_name": "John Doe",
      "end_station": "Semarang",
      "eta": "2020-08-20 00:00:00",
      "start_station": "Surabaya",
      "status": "Transit",
    },
  },
}
`;

exports[`action redux should handle TRIP_LIST_LOADING 1`] = `
Object {
  "add": Object {},
  "lists": Object {
    "response": undefined,
  },
}
`;

exports[`action redux should return initial state 1`] = `
Object {
  "add": Object {},
  "lists": Object {},
}
`;

exports[`action user redux should handle LOGIN 1`] = `
Object {
  "login": Object {
    "response": undefined,
  },
}
`;

exports[`action user redux should handle LOGIN_ERROR 1`] = `
Object {
  "login": Object {
    "error": Object {
      "error": "Username not found",
    },
  },
}
`;

exports[`action user redux should handle LOGIN_LOADING 1`] = `
Object {
  "login": Object {
    "loading": true,
  },
}
`;

exports[`action user redux should return initial state 1`] = `
Object {
  "login": Object {},
}
`;

exports[`reducer user redux should handle LOGIN 1`] = `
Object {
  "login": Object {
    "response": undefined,
  },
}
`;

exports[`reducer user redux should handle LOGIN_ERROR 1`] = `
Object {
  "login": Object {
    "error": Object {
      "error": "Username not found",
    },
  },
}
`;

exports[`reducer user redux should handle LOGIN_LOADING 1`] = `
Object {
  "login": Object {
    "loading": true,
  },
}
`;

exports[`reducer user redux should return initial state 1`] = `
Object {
  "login": Object {},
}
`;
